src/s3000_laser.cc

```diff
@@ -129,62 +138,61 @@ class s3000node {
 
   void ConnectTest(diagnostic_updater::DiagnosticStatusWrapper& status)
   {
-    laser->Open();
-    status.summary(0, "Connected successfully.");
+    if (laser->Open() == 0)
+      status.summary(0, "Connected successfully.");
+    else
+      status.summary(2, "Connection unsuccessful.");
   }
 
-  void getData(sensor_msgs::LaserScan& data)
+
+  int getData(sensor_msgs::LaserScan& data)
   {
     bool bValidData = false;
 
-    laser->ReadLaser( data, bValidData );
+    if (laser->ReadLaser( data, bValidData ) < 0)
+      return (-1);
 
     //// If valid data, publish it
-    if (bValidData) {
-    	data.header.stamp = ros::Time::now();
-	laser_data_pub_.publish( data );
-	}
-
-/*
-    // Robotnik - to test unit
-    if (publish_tf) {
-	// create a tf message
-	geometry_msgs::TransformStamped imu_trans;
-	imu_trans.header.stamp = data.header.stamp;
-	imu_trans.header.frame_id = "laser";   //odom_frame_id.c_str();
-	imu_trans.child_frame_id = "base_link";
-	imu_trans.transform.translation.x = 0.0;
-	imu_trans.transform.translation.y = 0.0;
-	imu_trans.transform.translation.z = 0.0;
-	imu_trans.transform.rotation = data.orientation;
-	//send the transform
-	imu_broadcaster.sendTransform(imu_trans);
-        }
-*/  
+    if (bValidData)
+    {
+      data.header.stamp = ros::Time::now();
+      laser_data_pub_.publish( data );
+      freq_diag_.tick();
+      getting_data_ = true;
+    }
+    else
+    {
+    	getting_data_ = false;
+    }
+
+    return (0);
+  }
 
-}
 
   bool spin()
   {
-
-    // sensor_msgs::LaserScan data;
     ros::Rate r(desired_freq_);
 
-    while (!ros::isShuttingDown()) // Using ros::isShuttingDown to avoid restarting the node during a shutdown.
+    // Using ros::isShuttingDown to avoid restarting the node during a shutdown.
+    while (!ros::isShuttingDown())
     {
       if (start() == 0)
       {
-        while(node_handle_.ok()) {
-          //if(publish_datum() < 0)	
-          //  break;
-          getData( reading );           
+        while(node_handle_.ok())
+        {
+          if (getData(reading) == -1) //Error reading from port
```

Comment:
Use `{...}` around the body of the if statement to avoid [goto fail](https://www.imperialviolet.org/2014/02/22/applebug.html)-style merge errors.